server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java

```diff
@@ -545,24 +547,33 @@ public TimeSeriesParams.MetricType getMetricType() {
 
         @Override
         public BlockLoader blockLoader(BlockLoaderContext blContext) {
-            if (blContext.fieldExtractPreference() == DOC_VALUES && hasDocValues()) {
```

Comment:
Wouldn't a smaller change be to just change:

```
 if ((blContext.fieldExtractPreference() == DOC_VALUES || isSyntheticSource)&& hasDocValues()) {
```

?

Reply:
wouldn't that break if the field has `store = true`? When stored is set, we store the point as its string representation and not long (see `index()` in GeoPointFieldMapper).